test/testrunner.py

```diff
@@ -158,15 +160,14 @@ def main():
         if specific_test_type == "sanity":
             pull_dlc_images(all_image_list)
         if specific_test_type == "eks":
-            eks_terminable_clusters = setup_eks_clusters(dlc_images)
+            new_eks_cluster_name = setup_eks_clusters(dlc_images)
         # Execute dlc_tests pytest command
         pytest_cmd = ["-s", "-rA", test_path, f"--junitxml={report}", "-n=auto"]
         try:
             sys.exit(pytest.main(pytest_cmd))
         finally:
-            if specific_test_type == "eks" and eks_terminable_clusters:
-                for cluster in eks_terminable_clusters:
-                    eks_utils.delete_eks_cluster(cluster)
+            if specific_test_type == "eks":
```

Comment:
Can you also add "new_eks_cluster_name" as a condition here? The reason being, why set it to None in the first place -- if it somehow sneaks through as None, we will try to call delete_eks_cluster on "None"